### PR TITLE
base: init-install-efi: installer: support encrypted rootfs

### PR DESCRIPTION
**Pull request by ldts**
Add support for a CI encrypted rootfs USB installer.

### CODE REVIEW

**Comment by ldts** [comment]
getting rid of "fiopassphrase" for the installer and the initramdisk should be done in a different commit.
having it first here allows users already using "fiopassphrase" to test the installer and TPM2 luks reencryption onf the next boot.

**Comment by ldts** [comment]
Enforce that the PK has been set before installing.

File: meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi/init-install-efi.sh
Comment (quaresmajose) [inline meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi/init-install-efi.sh:18]: the efivar tool should be more bullet proof instead of parsing binary strings.
It has no dependencies, so its installation size is super small.

```
# Whether the platform firmware is operating in Secure boot mode (1) or not (0)
efi_secure=`efivar --name=8be4df61-93ca-11d2-aa0d-00e098032b8c-SecureBoot --print-decimal`
# Whether the system should require authentication on SetVariable() requests to Secure Boot policy variables (0) or not (1)
efi_mode=`efivar --name=8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode --print-decimal`
```

The `8be4df61-93ca-11d2-aa0d-00e098032b8c` is a global variable and part of the spec
```
efivar --list-guids|grep 8be4df61-93ca-11d2-aa0d-00e098032b8c
{8be4df61-93ca-11d2-aa0d-00e098032b8c}  {global}        efi_guid_global EFI Global Variable
```

File: meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi_%.bbappend
Comment (quaresmajose) [inline meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi_%.bbappend:5]: `RDEPENDS:${PN} += "efivar"` to use the efivar tool on the ramdisk

Comment (ldts) [reply meta-lmp-base/recipes-core/initrdscripts/initramfs-module-install-efi_%.bbappend:5]: absolutely! makes sense - I shouldnt have followed the code that we have in LmP :) . will refactor that as well

**Comment by ldts** [comment]
@mike-scott please could you check https://foundriesio.atlassian.net/browse/FFTK-3548 
I think we need your go-ahead before merging this.

**Review by ricardosalveti — APPROVED** [review]
Fine with the changes, we need to see if we can find a way to test this as well with @mwasilew.

**Comment by mwasilew** [comment]
At the moment we're not testing the installer at all due to lack of some "preseed" support. Same applies to installing encrypted rootfs.

**Pull request merged**
